apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/RetryOptions.cs

```diff
@@ -0,0 +1,127 @@
+        public int RetryAttempts { get; }
+
+        /// <summary>The amount of time to wait between retries, only used for <see cref="RetryType.Retry"/>.</summary>
+        public TimeSpan RetryInterval { get; }
+
+        internal RetryOptions(RetryType retryType, ExceptionHandling exceptionHandling, BufferOverflow bufferOverflow,
+            int? bufferSizeBytes = null, int? retryAttempts = null, TimeSpan? retryInterval = null)
+        {
+            RetryType = GaxPreconditions.CheckEnumValue(retryType, nameof(retryType));
+            ExceptionHandling = GaxPreconditions.CheckEnumValue(exceptionHandling, nameof(exceptionHandling));
+            BufferOverflow = GaxPreconditions.CheckEnumValue(bufferOverflow, nameof(bufferOverflow));
+            BufferSizeBytes = GaxPreconditions.CheckArgumentRange(
+                bufferSizeBytes ?? 0, nameof(bufferSizeBytes), 0, int.MaxValue);
+            RetryAttempts = GaxPreconditions.CheckArgumentRange(
+                retryAttempts ?? 0, nameof(retryAttempts), 0, int.MaxValue);
+            RetryInterval = retryInterval ?? TimeSpan.Zero;
+            GaxPreconditions.CheckArgument(RetryInterval >= TimeSpan.Zero, nameof(retryInterval), 
+                $"{nameof(retryInterval)} must be greater than 0");
+        }
+
+        /// <summary>
+        /// Creates a <see cref="RetryOptions"/> for <see cref="RetryType.None"/>.
+        /// </summary>
+        /// <param name="exceptionHandling">Optional, how exceptions should be handled.
+        ///     Defaults to <see cref="ExceptionHandling.Ignore"/>.</param>
+        public static RetryOptions NoRetry(ExceptionHandling exceptionHandling = ExceptionHandling.Ignore)
+            => new RetryOptions(RetryType.None, exceptionHandling, BufferOverflow.IgnoreNewEntries);
+
+        /// <summary>
+        /// Creates a <see cref="RetryOptions"/> for <see cref="RetryType.Retry"/>.
+        /// </summary>
+        /// <param name="exceptionHandling">Optional, how exceptions should be handled.
+        ///     Defaults to <see cref="ExceptionHandling.Ignore"/>.</param>
+        /// <param name="bufferOverflow">Optional, how a buffer overflow should be handled.
+        ///     Defaults to <see cref="BufferOverflow.IgnoreNewEntries"/>.</param>
+        /// <param name="bufferSizeBytes">Optional, the size of the buffer to be used in bytes.
+        ///     Defaults to 2^18 = 262144.</param>
+        /// <param name="retryInterval">Optional, the time between retries. Default to 2 seconds.</param>
+        /// <param name="retryAttempts">Optional, the number of retry attempts. Defaults to 10.</param>
+        public static RetryOptions Retry(
+            ExceptionHandling exceptionHandling = ExceptionHandling.Ignore,
+            BufferOverflow bufferOverflow = BufferOverflow.IgnoreNewEntries,
+            int bufferSizeBytes = DefaultBufferSize,
```

Comment:
How certain are we of these defaults? For the enum values I think it's reasonable, but I'd be tempted to use an `int?` for `bufferSizeBytes` and initialize it like `retryInterval`, turning `DefaultBufferSize` into a `public static readonly int` instead, basically so we can change it later safely. (Ditto DefaultRetryAttempts.)